src/lib/storage/run_length_column/run_length_column_iterable.hpp

```diff
@@ -0,0 +1,134 @@
+
+    functor(begin, end);
+  }
+
+  template <typename Functor>
+  void _on_with_iterators(const ChunkOffsetsList& mapped_chunk_offsets, const Functor& functor) const {
+    auto begin = PointAccessIterator{*_column.values(), *_column.null_values(), *_column.end_positions(),
+                                     mapped_chunk_offsets.cbegin()};
+    auto end = PointAccessIterator{*_column.values(), *_column.null_values(), *_column.end_positions(),
+                                   mapped_chunk_offsets.cend()};
+
+    functor(begin, end);
+  }
+
+ private:
+  const RunLengthColumn<T>& _column;
+
+ private:
+  class Iterator : public BaseColumnIterator<Iterator, ColumnIteratorValue<T>> {
+   public:
+    using ValueIterator = typename pmr_vector<T>::const_iterator;
+    using NullValueIterator = typename pmr_vector<bool>::const_iterator;
+    using EndPositionIterator = typename pmr_vector<ChunkOffset>::const_iterator;
+
+   public:
+    explicit Iterator(const ValueIterator& value_it, const NullValueIterator& null_value_it,
+                      const EndPositionIterator& end_position_it, const ChunkOffset start_position)
+        : _value_it{value_it},
+          _null_value_it{null_value_it},
+          _end_position_it{end_position_it},
+          _current_position{start_position} {}
+
+   private:
+    friend class boost::iterator_core_access;  // grants the boost::iterator_facade access to the private interface
+
+    void increment() {
+      ++_current_position;
+
+      if (_current_position > *_end_position_it) {
+        ++_value_it;
+        ++_null_value_it;
+        ++_end_position_it;
+      }
+    }
+
+    bool equal(const Iterator& other) const { return _current_position == other._current_position; }
+
+    ColumnIteratorValue<T> dereference() const {
+      return ColumnIteratorValue<T>{*_value_it, *_null_value_it, _current_position};
+    }
+
+   private:
+    ValueIterator _value_it;
+    NullValueIterator _null_value_it;
+    EndPositionIterator _end_position_it;
+    ChunkOffset _current_position;
+  };
+
+  class PointAccessIterator : public BasePointAccessColumnIterator<PointAccessIterator, ColumnIteratorValue<T>> {
+   public:
+    explicit PointAccessIterator(const pmr_vector<T>& values, const pmr_vector<bool>& null_values,
+                                 const pmr_vector<ChunkOffset>& end_positions,
+                                 const ChunkOffsetsIterator& chunk_offsets_it)
+        : BasePointAccessColumnIterator<PointAccessIterator, ColumnIteratorValue<T>>{chunk_offsets_it},
+          _values{values},
+          _null_values{null_values},
+          _end_positions{end_positions},
+          _prev_chunk_offset{end_positions.back() + 1u},
+          _prev_index{end_positions.size()} {}
+
+   private:
+    friend class boost::iterator_core_access;  // grants the boost::iterator_facade access to the private interface
+
+    ColumnIteratorValue<T> dereference() const {
```

Comment:
Without comments, this method is hard to understand even for people who have heard of RLE before.

Reply:
done